packages/vite/src/node/server/index.ts

```diff
@@ -526,6 +527,8 @@ export async function _createServer(
     hot: createDeprecatedHotBroadcaster(ws),
 
     environments,
+    ...environments,
```

Comment:
On restart, we probably need to cleanup removed environments from the `server` variable.

Reply:
I think this isn't needed, we are now doing
```js
    _setInternalServer(_server: ViteDevServer) {
      // Rebind internal the server variable so functions reference the user
      // server instance after a restart
      server = _server
    },
```